packages/plugins/Gfi/src/store/actions/debouncedGfiRequest.ts

```diff
@@ -0,0 +1,159 @@
+    }
+
+    const layerConfiguration = configuration.gfi?.layers[key] || {}
+    const layerSpecification = rawLayerList.getLayerWhere({ id: key })
+    const mainLayerConfiguration = configuration.layers.find(
+      (element) => element.id === key
+    )
+    const layerGfiMode =
+      mainLayerConfiguration?.gfiMode || configuration?.gfi?.mode || 'bboxDot'
+
+    return requestGfi({
+      map,
+      layer,
+      coordinate,
+      layerConfiguration,
+      layerSpecification,
+      mode: layerGfiMode,
+    })
+  })
+
+const filterFeatures = (
+  featuresByLayerId: Record<
+    string,
+    symbol | GeoJsonFeature<GeoJsonGeometry, GeoJsonProperties>[]
+  >
+): Record<string, GeoJsonFeature<GeoJsonGeometry, GeoJsonProperties>[]> => {
+  const entries = Object.entries(featuresByLayerId)
+  const filtered = entries.filter((keyValue) => Array.isArray(keyValue[1])) as [
+    string,
+    GeoJsonFeature<GeoJsonGeometry, GeoJsonProperties>[]
+  ][]
+  return Object.fromEntries(filtered)
+}
+
+/**
+ * Code from `getFeatureInfo`, pulled to avoid overly requesting feature
+ * information. Since sources in Pins plugin update right after each other
+ * (and such effects are to be expected across the system), we're debouncing
+ * this *after* resetting the module state, as something is bound to happen.
+ */
+// eslint-disable-next-line max-lines-per-function
+const gfiRequest =
+  (featureDisplayLayer: VectorLayer<Feature<Geometry>>) =>
+  async (
+    actionContext: PolarActionContext<GfiState, GfiGetters>,
+    coordinate: [number, number]
+  ): Promise<void> => {
+    const {
+      commit,
+      rootGetters: { map },
+      getters: { geometryLayerKeys, afterLoadFunction },
+    } = actionContext
+    // fetch new feature information for all configured layers
+    const promisedFeatures = getPromisedFeatures(actionContext, coordinate)
+
+    const errorSymbol = (err) => Symbol(err)
+    const features = (await Promise.allSettled(promisedFeatures)).map(
+      (result) =>
+        result.status === 'fulfilled'
+          ? result.value
+          : errorSymbol(result.reason.message)
+    )
+
+    const srsName: string = map.getView().getProjection().getCode()
+    let featuresByLayerId: Record<string, GeoJsonFeature[] | symbol> =
+      mapFeaturesToLayerIds(actionContext, features, srsName)
+
+    // store features in state, if configured via client after specific function
+    if (typeof afterLoadFunction === 'function') {
+      featuresByLayerId = await afterLoadFunction(
+        filterFeatures(featuresByLayerId),
+        srsName
+      )
+    }
+    commit('setFeatureInformation', featuresByLayerId)
+
+    // render feature geometries to help layer
+    geometryLayerKeys
+      .filter((key) => Array.isArray(featuresByLayerId[key]))
+      .forEach((key) =>
+        // @ts-expect-error | Might be fixed through having all the types in the action. Otherwise: It works properly, as all the symbols are filtered before calling forEach
```

Comment:
The comment regarding the `@ts-expect-error` isn't true anymore. We can leave like this, with an adjusted comment

Reply:
https://github.com/Dataport/polar/pull/176/commits/5cf9161b46fe7655119116a79234bab477cdba94